### ISSUE TITLE
Mode for key_metric

### ISSUE THREAD

**Issue by ChenglongWang** [Project-MONAI/MONAI#1113]
**Is your feature request related to a problem? Please describe.**
I think the Workflow or CheckpointSaver need a `mode` option for key_metric and additional_metrics to handle both metrics like `acc` and `mse`.

**Describe the solution you'd like**
Here is the keras version:
```
if mode == 'min':
      self.monitor_op = np.less
      self.best = np.Inf
elif mode == 'max':
      self.monitor_op = np.greater
      self.best = -np.Inf
else: #auto mode
      if 'acc' in self.monitor or self.monitor.startswith('fmeasure'):
        self.monitor_op = np.greater
        self.best = -np.Inf
      else:
        self.monitor_op = np.less
        self.best = np.Inf
```

**Comment by Nic-Ma** [comment]
Hi @ChenglongWang ,

Thanks for your interest in MONAI workflows.
We developed the workflows to compare metrics referring to ignite @vfdev-5 :
https://github.com/pytorch/ignite/blob/master/ignite/handlers/checkpoint.py#L320
May I know in which case to set mode `less`? I think ACC and MSE are in `greater` mode?

Thanks.

**Comment by vfdev-5** [comment]
As far as I understand, the feature request is to provide 2 + autor ways to compare metrics to decide what to save. In ignite we assume to save best score as larger score and for metrics like MSE we ask user to provide inverse of it such that the best model has larger score.

**Comment by ChenglongWang** [comment]
@Nic-Ma @vfdev-5 Thanks for your reply. 
Is there any example of providing inverse of metric to solve my current issue. I cannot find a solution in MONAI workflow style.
Here gives my evaluator example:
```
evaluator = SupervisedEvaluator(
       device=device,
       val_data_loader=test_loader,
       network=net,
       inferer=SimpleInferer(),
       key_val_metric={
           "val_mse": MeanSquaredError()
       },
       val_handlers=[CheckpointSaver(save_dir=model_dir, save_dict={"net": net}, save_key_metric=True, key_metric_name='val_mse')],
       amp=opts.amp
 )
```

**Comment by Nic-Ma** [comment]
Hi @vfdev-5 ,

Could you please help share an ignite example? MONAI metrics just extend the ignite metric base class.

Thanks.

**Comment by vfdev-5** [comment]
Hi @Nic-Ma here is an example of usage of ignite's Checkpoint with `score_function` to adapt mse as "increasing score" :

```python
from ignite.engine import Engine, Events
from ignite.handlers import Checkpoint, DiskSaver, global_step_from_engine

trainer = ...
evaluator = ...

def score_function(engine):
    return -1.0 * engine.state.metrics['mse']

to_save = {'model': model}
handler = Checkpoint(
    to_save, 
    DiskSaver('/tmp/models', create_dir=True), 
    n_saved=2,
    filename_prefix='best', 
    score_function=score_function, 
    score_name="neg_val_mse",
    global_step_transform=global_step_from_engine(trainer)
)
evaluator.add_event_handler(Events.COMPLETED, handler)

> ["best_model_9_neg_val_mse=-0.15.pt", "best_model_10_neg_val_mse=-0.08.pt", ]
```

HTH
